code/modules/multiz/hoist.dm

```diff
@@ -0,0 +1,292 @@
+///////////////////////////
+// Dost thou even hoist? //
+///////////////////////////
+
+#define NORMAL_LAYER 3
+
+/obj/item/hoist_kit
+	name = "hoist kit"
+	desc = "A setup kit for a hoist that can be used to lift things. The hoist will deploy in the direction you're facing."
+	icon = 'icons/obj/hoists.dmi'
+	icon_state = "hoist_case"
+
+/obj/item/hoist_kit/attack_self(mob/user)
+	if (!do_after(usr, (2 SECONDS), src))
+		return
+
+	new /obj/structure/hoist(get_turf(user), user.dir)
+	user.visible_message(
+		SPAN_WARNING("[user] deploys the hoist kit!"),
+		SPAN_NOTICE("You deploy the hoist kit!"),
+		SPAN_NOTICE("You hear the sound of parts snapping into place."))
+	qdel(src)
+
+/obj/effect/hoist_hook
+	name = "hoist clamp"
+	desc = "A clamp used to lift people or things."
+	icon = 'icons/obj/hoists.dmi'
+	icon_state = "hoist_hook"
+	can_buckle = TRUE
+	anchored = TRUE
+
+	var/obj/structure/hoist/source_hoist
+
+/obj/effect/hoist_hook/attack_hand(mob/living/user)
+	return // no, bad
+
+/obj/effect/hoist_hook/MouseDrop_T(atom/movable/AM,mob/user)
+	if (issilicon(user))
+		return
+
+	if (!AM.simulated || AM.anchored)
+		to_chat(user, SPAN_NOTICE("You can't do that."))
+		return
+	if (source_hoist.hoistee)
+		to_chat(user, SPAN_NOTICE("\The [source_hoist.hoistee] is already attached to \the [src]!"))
+		return
+	source_hoist.attach_hoistee(AM)
+	user.visible_message(
+		SPAN_DANGER("[user] attaches \the [AM] to \the [src]."),
```

Comment:
danger should be reserved for direct combat danger, this is notice probably, warning at most